api/lib/infrastructure/repositories/user-repository.js

```diff
@@ -244,20 +243,28 @@ module.exports = {
       });
   },
 
-  async updateUserDetailsForAdministration(id, userAttributes) {
-    try {
-      const updatedUser = await BookshelfUser.where({ id }).save(userAttributes, { patch: true, method: 'update' });
-      await updatedUser.related('authenticationMethods').fetch({ require: false });
-      return _toUserDetailsForAdminDomain(updatedUser);
-    } catch (err) {
-      if (err instanceof BookshelfUser.NoRowsUpdatedError) {
-        throw new UserNotFoundError(`User not found for ID ${id}`);
-      }
-      if (isUniqConstraintViolated(err)) {
-        throw new AlreadyExistingEntityError('Cette adresse e-mail ou cet identifiant est déjà utilisé(e).');
-      }
-      throw err;
+  async updateUserDetailsForAdministration(id, updatedUserAttributes) {
+    const query = knex('users');
+    if (updatedUserAttributes?.email) {
+      query.orWhere('email', updatedUserAttributes?.email);
+    }
+    if (updatedUserAttributes?.username) {
+      query.orWhere('username', updatedUserAttributes?.username);
+    }
+    const [user] = await query;
+    if (user) {
+      throw new AlreadyExistingEntityError('Cette adresse e-mail ou cet identifiant est déjà utilisé(e).');
```

Comment:
Suggestion : faire un extract method de ce bout de code dans une fonction `
`checkIfEmailOrUsernameAlreadyUsed`.

Reply:
Je pense que ce n'est pas utile de le sortir vu que c'est uniquement cette méthode `updateUserDetailsForAdministration` qui l'utilise. De plus ce n'est pas un check, je dirais que c'est plutôt une construction de requête db.

Reply:
Je pensais méthode privée, je sais que l'équipe fait pas trop ça mais je trouve plus clair de faire des petites fonctions qui décrivent ce qu'elles font. C'est plus facile je trouve de revenir 6 mois plus tard sur le bout de code et de lire un nom de méthode pour comprendre le métier.
Tu dis que c'est une construction de query mais je vois pas d'utilisation ensuite de user ou de query 
`const [user] = await query;`
On dirait vraiment que ça ne sert qu'à générer une erreur ou rien (d'où le fait que je dise que c'est un check).

Je viens de regarder, il y a une exemple dans le usecase updateUserDetailsForAdministration  api/lib/domain/usecases/update-user-details-for-administration.js qui fait appel à ce repo (y'a même double check du coup...)

Fonctionnellement, c'est une erreur qui apparaît quand on a anonymisé l'user et qu'on reclicke pour anonymiser à nouveau.
Pour l'update des détails de l'utilisateur,